src/lib/pages/evm-contract-details/data.ts

```diff
@@ -0,0 +1,74 @@
+import { useEffect, useMemo, useState } from "react";
+
+import {
+  useEvmTxHashesByCosmosTxHashes,
+  useTxsByAddressSequencer,
+  useTxsDataJsonRpc,
+} from "lib/services/tx";
+import type { TxDataWithTimeStampJsonRpc } from "lib/services/types";
+import type { BechAddr20 } from "lib/types";
+
+export const useContractDetailsTxs = (address: BechAddr20) => {
+  const [evmTxs, setEvmTxs] = useState<TxDataWithTimeStampJsonRpc[]>();
+  const [isEvmTxsLoading, setIsEvmTxsLoading] = useState(true);
+  const [isEvmTxsFetching, setIsEvmTxsFetching] = useState(false);
+
+  const cosmosTxHashToEvmTx = useMemo(
+    () => new Map<string, TxDataWithTimeStampJsonRpc>(),
+    []
+  );
+
+  const {
+    data: cosmosTxs,
+    fetchNextPage,
+    hasNextPage,
+    isFetching: isCosmosTxsLoading,
+    isFetchingNextPage: isCosmosTxsFetchingNextpage,
+    latestFetchedData: newCosmosTxs,
+  } = useTxsByAddressSequencer(address, undefined);
+
+  const { data: newEvmTxHashes, isFetching: isEvmHashesFetching } =
+    useEvmTxHashesByCosmosTxHashes(newCosmosTxs?.map((tx) => tx.hash));
+  const { data: evmTxsData, isFetching: isEvmTxsDataFetching } =
+    useTxsDataJsonRpc(newEvmTxHashes?.filter((tx) => tx !== null) as string[]);
+
+  useEffect(() => {
+    setIsEvmTxsFetching(true);
+    if (!newCosmosTxs || !newEvmTxHashes || !evmTxsData) return;
```

Comment:
Swap the following lines because `setIsEvmTxsFetching(true)` should be called only after confirming that `newCosmosTx`, `newEvmTxHash`, and `evmTxsData` are defined. Otherwise, setting `isEvmTxsFetching` to true will not make sense if we immediately return due to missing data.